src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java

```diff
@@ -534,6 +534,37 @@ private boolean hasEmptyLineBefore(DetailAST token) {
     }
 
     /**
+     * Check if token is predicted by javadoc comment
+     * @param token token.
+     * @return true, if token is predicted by javadoc comment.
+     */
+    private boolean isPredictedByJavadoc(DetailAST token) {
+        final int lineNo = token.getLineNo();
+        boolean result = false;
+
+        if (lineNo != 1) {
+            //  [lineNo - 2] is the number of the previous line because the numbering starts
+            //  from zero.
+            String lineBefore;
+            int lineIndex = 2;
+            do {
+                lineBefore = getLines()[lineNo - lineIndex].trim();
```

Comment:
why not just make `lineIndex` the value of `lineNo` - 2 and decrement each loop. This will save us from having to redo this calculation each time through, twice.

Also please stash the result of `getLines` as we do a clone on the values each call.